File: mobile/library/kotlin/io/envoyproxy/envoymobile/stats/Distribution.kt
package io.envoyproxy.envoymobile

/** A time-series distribution that tracks quantile/sum/average stats. */
interface Distribution {

  /** Records a new value to add to the distribution. */
  fun recordValue(value: Int)

  /** Records a new value to add to the distribution along with the tags. */
  fun recordValue(tags: Tags = TagsBuilder().build(), value: Int)
}

File: mobile/library/swift/stats/Distribution.swift
import Foundation

/// A time series distribution tracking quantile/sum/average stats.
@objc
public protocol Distribution: AnyObject {
  /// Record a new value to add to the integer distribution.
  ///
  /// - parameter value: The value to record.
  func recordValue(value: Int)

  /// Record a new value to add to the integer distribution with tags.
  ///
  /// - parameter tags:  The tags to attach to this distribution.
  /// - parameter value: The value to record.
  func recordValue(tags: Tags, value: Int)
}
